File: akka-serialization-jackson/src/test/scala/doc/akka/serialization/jackson/v2a/Address.scala
/*
 * Copyright (C) 2019-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package doc.akka.serialization.jackson.v2a

import akka.serialization.jackson.JsonSerializable

// #structural
case class Address(street: String, city: String, zipCode: String, country: String) extends JsonSerializable
// #structural

File: akka-serialization-jackson/src/test/java/jdoc/akka/serialization/jackson/v2a/Address.java
/*
 * Copyright (C) 2019-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package jdoc.akka.serialization.jackson.v2a;

// #structural
public class Address {
  public final String street;
  public final String city;
  public final String zipCode;
  public final String country;

  public Address(String street, String city, String zipCode, String country) {
    this.street = street;
    this.city = city;
    this.zipCode = zipCode;
    this.country = country;
  }
}
// #structural
